src/util/bit_util.rs

```diff
@@ -207,7 +207,7 @@ impl BitWriter {
   #[inline]
   pub fn skip(&mut self, num_bytes: usize) -> Result<usize> {
     self.flush();
-    assert!(self.byte_offset < self.max_bytes);
+    assert!(self.byte_offset <= self.max_bytes);
```

Comment:
This seems to be the correct check, arrow bit writer has assertion on `less than or equal`.

Reply:
Yes this looks right.